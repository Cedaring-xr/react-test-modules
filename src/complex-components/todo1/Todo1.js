import React from 'react'

export default function TodoList1({ todo }) {
    const { id, title, completed } = todo
    const p = <p>{title}</p>
    const text = completed ? <strike>{p}</strike> : p
    


  return (
    <div data-testid={`todo-${id}`}>{ text }</div>
  )
}
