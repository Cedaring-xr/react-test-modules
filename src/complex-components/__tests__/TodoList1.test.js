import { render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import Todo1 from '../todo1/Todo1'
import renderer from 'react-test-renderer'

afterEach(() => {
    cleanup()
})

test('should render non-completed todo', () => {
    const mockTodo = {id: 2, title: 'test task 1', completed: false}
    render(<Todo1 todo={mockTodo}/>)
    const element = screen.getByTestId('todo-2')
    expect(element).toBeDefined();
    expect(element).not.toContainHTML('strike')
})

test('should render completed todo', () => {
    const mockTodo = {id: 2, title: 'test task 2', completed: true}
    render(<Todo1 todo={mockTodo}/>)
    const element = screen.getByTestId('todo-2')
    expect(element).toBeDefined();
    expect(element).toContainHTML('strike')
})

// snapshot component testing
test('should match snapshot of TodoList', () => {
    const mockTodo = {id: 2, title: 'test task 2', completed: true}
    const tree = renderer.create(<Todo1 todo={mockTodo}/>).toJSON()
    expect(tree).toMatchSnapshot()
})