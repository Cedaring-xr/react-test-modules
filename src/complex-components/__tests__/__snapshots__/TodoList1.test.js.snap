// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot of TodoList 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
    <p>
      test task 2
    </p>
  </strike>
</div>
`;
